File: 프로그래머스/lv2/12973. 짝지어 제거하기/짝지어 제거하기.js
function solution(s){
    
    while(true) {
       let count = 0;
       let arr = [];
       for(let i = 0; i < s.length; i++) {
            if(arr.length === 0) {
                arr.push(s[i])
            } else {
               if(arr[arr.length - 1] === s[i]) {
                arr.pop()
                count++;
                if(s.length === 0) break;
              } else {
                arr.push(s[i])
             } 
            } 
        }
        s = arr.join('');
        if(count === 0) break;
    }
    
    return s.length === 0 ? 1 : 0;
}

File: 프로그래머스/lv2/12973. 짝지어 제거하기/짝지어 제거하기.java
import java.util.*;
import java.lang.*;
class Solution{
    public int solution(String s){
       String[] strArr = s.split("");
       while(true) {
           int count = 0;
           Stack<String> stack = new Stack<>();
           if(strArr.length == 0) break;
           for(int i = 0; i < strArr.length; i++) {
               if(stack.size() == 0) {
                   stack.add(strArr[i]);
               } else {
                   if(stack.peek().equals(strArr[i])) {
                      stack.pop();
                      count++;
                   } else {
                      stack.add(strArr[i]);
                   }
               }
           }
           strArr = Arrays.stream(stack.toArray()).toArray(String[]::new);
           if(count == 0) break;
       }
           
     return strArr.length == 0 ? 1 : 0;
    }
}